rust/src/runner/context.rs

```diff
@@ -228,3 +228,20 @@ impl TestContext {
         thread::sleep(self.naptime)
     }
 }
+
+// We implement Drop to circumvent the errors from unlinking a directory for which
+// search or write permission is denied.
+impl Drop for TestContext {
+    fn drop(&mut self) {
+        let iter = walkdir::WalkDir::new(self.base_path()).into_iter();
```

Comment:
Is this more or less what `/bin/rm -r` does?

Reply:
It doesn't remove, but it changes the permissions to allow the destructor of `TempDir` to succeed.

Reply:
We could maybe write our own `TempDir`?

Reply:
This is equivalent to `chmod -R 700`, which is much more work than necessary.  You should actually only need to change the permissions of directories.  Files can be unlinked even without write permission.  The returned `entry` caches the file type, so you should check that, and skip the chmod if it isn't a directory.
You may also need to `chflags 0` some files before deleting them, if their SF_IMMUTABLE or UF_IMMUTABLE bit is set.

Reply:
Thanks, I added the checks (for stat and file type), and `chflags` to reset the flags.